device/pci_device.hpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * SPDX-FileCopyrightText: (c) 2023 Tenstorrent Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+#include <cstdint>
+#include <vector>
+
+#include "device/tt_arch_types.h"
+#include "architecture_implementation.h"
+
+static uint32_t GS_BAR0_WC_MAPPING_SIZE = (156<<20) + (10<<21) + (18<<24);
+static uint32_t BH_BAR0_WC_MAPPING_SIZE = 188<<21; // Defines the address for WC region. addresses 0 to BH_BAR0_WC_MAPPING_SIZE are in WC, above that are UC
+
+static const uint32_t BH_NOC_NODE_ID_OFFSET = 0x1FD04044;
+static const uint32_t GS_WH_ARC_SCRATCH_6_OFFSET = 0x1FF30078;
+
+// See /vendor_ip/synopsys/052021/bh_pcie_ctl_gen5/export/configuration/DWC_pcie_ctl.h
+static const uint64_t UNROLL_ATU_OFFSET_BAR = 0x1200;
+
+// BAR0 size for Blackhole, used to determine whether write block should use BAR0 or BAR4
+const uint64_t BAR0_BH_SIZE = 512 * 1024 * 1024;
+
+// PCIe device IDs through ioctl
+static const uint16_t GS_PCIE_DEVICE_ID = 0xfaca;
```

Comment:
@pjanevskiTT @broskoTT 
Constants for pcie device ids. Please let me know if these const are sufficient

Reply:
It's fine